packages/svelte/src/internal/client/reactivity/sources.js

```diff
@@ -172,6 +218,15 @@ export function internal_set(source, value) {
 		source.v = value;
 		source.wv = increment_write_version();
 
+		var onchange = source.o?.onchange;
```

Comment:
This is the wrong place, it needs to happen just before we trigger inspect effects below, otherwise the versioning will be wrong

Reply:
Fixed it in the latest commit

Reply:
Come to think about it, should it happen after inspect effects or not? What if you change the value again in `onchange`?

Reply:
Correct...fixed it